DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+
+namespace DataStructures.Lists 
+{
+    public class CircularBuffer<T> 
+    {
+        /// <summary>
+        /// Initializes a circular buffer with initial length of 10
+        /// </summary>
+        public CircularBuffer(bool canOverride=true) : this(_defaultBufferLength, canOverride) 
+        {
+        }
+
+        /// <summary>
+        /// Initializes a circular buffer with given length
+        /// </summary>
+        /// <param name="length">The length of the buffer</param>
+        public CircularBuffer(int length, bool canOverride=true) 
+        {
+            if (length < 1) 
+            {
+                throw new ArgumentOutOfRangeException("length can not be zero or negative");
+            }
+            _circularBuffer = new T[length + 1];
+            _end = 0;
+            _start = 0;
+            CanOverride = canOverride;
+        }
+
+        /// <summary>
+        /// Writes value to the back of the buffer
+        /// </summary>
+        /// <param name="value">value to be added to the buffer</param>
+        public void Write(T value) 
```

Comment:
The use of the `Write` word is not appropriate, if I use the `List.Write()` method, it will make me believe I am re-writing the entire lists contents with that argument, a more suitable name would be `Add`, since that's what the method is doing.

Reply:
Yeah. It makes absolute sense.

Done.